src/main/java/io/r2dbc/postgresql/codec/OffsetTimeCodec.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.postgresql.codec;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.r2dbc.postgresql.client.Parameter;
+import io.r2dbc.postgresql.message.Format;
+import io.r2dbc.postgresql.type.PostgresqlObjectId;
+import io.r2dbc.postgresql.util.Assert;
+import io.r2dbc.postgresql.util.ByteBufUtils;
+
+import java.time.OffsetTime;
+
+import static io.r2dbc.postgresql.message.Format.FORMAT_TEXT;
+import static io.r2dbc.postgresql.type.PostgresqlObjectId.TIMETZ;
+
+public class OffsetTimeCodec extends AbstractTemporalCodec<OffsetTime> {
+
+    private final ByteBufAllocator byteBufAllocator;
+
+    OffsetTimeCodec(ByteBufAllocator byteBufAllocator) {
+        super(OffsetTime.class);
+        this.byteBufAllocator = Assert.requireNonNull(byteBufAllocator, "byteBufAllocator must not be null");
+    }
+
+    @Override
+    PostgresqlObjectId getDefaultType() {
+        return TIMETZ;
+    }
+
+    @Override
+    OffsetTime doDecode(ByteBuf buffer, PostgresqlObjectId dataType, Format format, Class<? extends OffsetTime> type) {
+        Assert.requireNonNull(buffer, "byteBuf must not be null");
+
+        return decodeTemporal(buffer, dataType, format, OffsetTime.class, OffsetTime::from);
+    }
+
+    @Override
+    Parameter doEncode(OffsetTime value) {
+        Assert.requireNonNull(value, "value must not be null");
+
+        return create(TIMETZ, FORMAT_TEXT, () -> ByteBufUtils.encode(this.byteBufAllocator, value.toString()));
```

Comment:
Would it make sense to encode `OffsetTime` always binary to preserve `offsetSec`?

Reply:
hmmmm... I'm not sure this comment is appropriate as I don't exactly know what FORMAT_TEXT means. Assuming it means the way the data is transmitted. If you go to binary there will be no offset. Postgres does not store offsets in the column. All data is stored in UTC and then the offset is whatever the client sets it to

Reply:
Text is the way data is represented on the wire. In this case, please disregard my earlier comment. I somehow lived under the impression that binary had more details than the text representation.

Reply:
Ironically it has less :)